jira_agile_metrics/calculators/cycletime.py

```diff
@@ -237,25 +235,28 @@ def calculate_cycle_times(
                 impediment_flag = None
                 impediment_start = None
                 impediment_start_status = None
-            
-            # Wipe timestamps if items have moved backwards; calculate cycle time
+
+            # calculate cycle time
 
             previous_timestamp = None
-            accepted_timestamp = None
-            completed_timestamp = None
+            committed_timestamp = None
+            done_timestamp = None
 
-            for cycle_name in cycle_names:
+            for cycle_name in reversed(cycle_names):
                 if item[cycle_name] is not None:
                     previous_timestamp = item[cycle_name]
 
-                    if accepted_timestamp is None and previous_timestamp is not None and cycle_name in accepted_steps:
-                        accepted_timestamp = previous_timestamp
-                    if completed_timestamp is None and previous_timestamp is not None and cycle_name in completed_steps:
-                        completed_timestamp = previous_timestamp
+                if previous_timestamp is not None:
+                    item[cycle_name] = previous_timestamp
```

Comment:
What is this line doing?

I am probably being a bit slow but I'm confused by the idea that we're going backwards through the cycle, and then setting the timestamp at the current cycle step to the timestamp of the "previous" in the iteration (which presumably is the *next* in the cycle)?

Can you explain the logic?

Reply:
Imagine workflow `A`, `B`, `C` and were going backwards starting at `C`. If `C` has a timestamp and `B` doesn't because the issue went straight from `A` to `C`, then I can use the timestamp from `C` for `B` too. This is semantically correct, because it means that the issue spent zero seconds in `B`, and it makes calculations easier. This is equal to the way ActionableAgile Analytics handels cases with skipped columns.